flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -332,6 +334,27 @@ public String toString() {
 		return taskInfo.build();
 	}
 
+	/**
+	 * Get the port keys representing the TM's configured endpoints. This includes mandatory TM endpoints such as
+	 * data and rpc as well as optionally configured endpoints for services such as prometheus reporter
+	 *
+	 * @return A deterministicly ordered Set of port keys to expose from the TM container
+	 * @param config
+	 */
+	static Set<String> extractPortKeys(Configuration config) {
+		LinkedHashSet<String> tmPortKeys = new LinkedHashSet<>(Arrays.asList(TM_PORT_KEYS));
+
+		if (config.containsKey(PORT_ASSIGNMENT_KEY)) {
+			String portAssignments = config.getString(PORT_ASSIGNMENT_KEY, "");
+			Arrays.stream(portAssignments.split(","))
+				.map(String::trim)
+				.peek(key -> LOG.debug("Adding port key " + key + " to mesos request"))
```

Comment:
Slf4j place holder should be preferred to avoid String concatenation, i.e., `LOG.debug("Adding port key {} to mesos request", key)`